src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -94,16 +94,16 @@
     # The strange regex here it to make sure we don't accidentally match numeric literals. We
     # also use a regex to explicitly exclude disallowed keywords.
     NakedIdentifierSegment=ReSegment.make(
-        r"[A-Z0-9_]*[A-Z][A-Z0-9_]*", name='identifier', type='naked_identifier',
+        r"[A-Z0-9_]*[A-Z][A-Z0-9_]*", name='naked_identifier', type='identifier',
         _anti_template=r"^(SELECT|JOIN|ON|USING|CROSS|INNER|LEFT|RIGHT|OUTER|INTERVAL|CASE|FULL|NULL)$"),
     FunctionNameSegment=ReSegment.make(r"[A-Z][A-Z0-9_]*", name='function_name', type='function_name'),
     # Maybe data types should be more restrictive?
     DatatypeIdentifierSegment=ReSegment.make(r"[A-Z][A-Z0-9_]*", name='data_type_identifier', type='data_type_identifier'),
     # Maybe date parts should be more restrictive
     DatepartSegment=ReSegment.make(r"[A-Z][A-Z0-9_]*", name='date_part', type='date_part'),
-    QuotedIdentifierSegment=NamedSegment.make('double_quote', name='identifier', type='quoted_identifier'),
-    QuotedLiteralSegment=NamedSegment.make('single_quote', name='literal', type='quoted_literal'),
-    NumericLiteralSegment=NamedSegment.make('numeric_literal', name='literal', type='numeric_literal'),
+    QuotedIdentifierSegment=NamedSegment.make('double_quote', name='quoted_identifier', type='identifier'),
+    QuotedLiteralSegment=NamedSegment.make('single_quote', name='quoted_literal', type='literal'),
+    NumericLiteralSegment=NamedSegment.make('numeric_literal', name='numeric_literal', type='literal'),
     TrueSegment=KeywordSegment.make('true', name='true', type='boolean_literal'),
```

Comment:
Given that the the QuotedIdentifierSegment and NumericLiteralSegment have changed the name and type, should the TrueSegment and FalseSegment also change?

Reply:
Yes - very good point

Reply:
I've addressed this with some of the other commits below.